template/example_test.go

```diff
@@ -0,0 +1,61 @@
+package template_test
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"go.expect.digital/mf2/template"
+)
+
+func ExampleTemplate_simpleMessage() {
+	// Define a MF2 string.
+	const input = "Hello, { $firstName :upper } { $lastName :lower style=first }!"
+
+	// Parse template.
+	t, err := template.New().Parse(input)
+	if err != nil {
+		// Handle error.
+	}
+
+	// Add functions to the template.
+	t.AddFunc("upper", func(operand any, _ map[string]any) (string, error) {
+		return strings.ToUpper(fmt.Sprint(operand)), nil
+	})
+
+	t.AddFunc("lower", func(operand any, options map[string]any) (string, error) {
+		if options == nil {
+			return strings.ToLower(fmt.Sprint(operand)), nil
+		}
+
+		if style, ok := options["style"].(string); ok && style == "first" {
+			return strings.ToLower(fmt.Sprint(operand)[0:1]) + fmt.Sprint(operand)[1:], nil
+		}
+
+		return "", fmt.Errorf("bad options")
+	})
```

Comment:
`lower` does not retain behaviour symmetry to `upper`.

I suggest to use other function with options - e.g. `trim` https://cs.opensource.google/go/go/+/go1.22.0:src/strings/strings.go;l=891

Reply:
This is just example, so it will be fine to leave as is